platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java

```diff
@@ -38,4 +38,16 @@ public class TransactionUtils {
             throw new UncheckedIOException("Could not parse transaction kind:%s".formatted(bytes.toHex()), e);
         }
     }
+
+    public static boolean isSystemTransaction(@NonNull final Bytes bytes) {
+        final SerializableDataInputStream in = new SerializableDataInputStream(bytes.toInputStream());
+
+        try {
```

Comment:
It might be performance heavy to try to serialize every transaction that is incoming with having a try/catch block. You can add a marker byte as @lpetrovic05 suggested in another PR.